runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsOverlay.java

```diff
@@ -149,15 +161,20 @@ public Dimension render(Graphics2D graphics)
 					Item item = (Item) current;
 					int itemId = item.getId();
 					int itemQuantity = item.getQuantity();
+					// Item does not support getting the name, so we need to get the ItemComposition instead
+					ItemComposition itemDefinition = client.getItemDefinition(itemId);
 
 					Integer currentQuantity = items.get(itemId);
-					if (currentQuantity == null)
-					{
-						items.put(itemId, itemQuantity);
-					}
-					else
+					if (!hiddenItems.contains(itemDefinition.getName()))
```

Comment:
Using continue there breaks the client, it never advances to the next item to put into the arraylist.

artificially throwing a current = current.getNext() and then continuing would probably work though.